### PR TITLE
feat: cfd 454 view point to point product details

### PR DESCRIPTION
**Pull request by matt-twomey**
# Description

-   View details for point to point products

# Testing instructions

-   Create one of each ticket listed in CFD-454 jira ticket and ensure correct fields are present for each

# Type of change

-   [x] feat - A new feature
-   [ ] fix - A bug fix
-   [ ] docs - Documentation only changes
-   [ ] codestyle - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] refactor - A code change that neither fixes a bug nor adds a feature
-   [ ] perf - A code change that improves performance
-   [ ] test - Adding missing tests or correcting existing tests
-   [ ] build - Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
-   [ ] ci - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] chore - Other changes that don't modify src or test files
-   [ ] revert - Reverts a previous commit
-   [ ] content - Changes to content or copy

### CODE REVIEW

File: repos/fdbt-site/src/constants/featureFlag.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
+// const dev = 'dev' === process.env.STAGE;
 const devOrTest = ['test', 'dev'].includes(process.env.STAGE ?? '');
 
 export const globalSettingsEnabled = true;
 export const globalSettingsDeleteEnabled = devOrTest;
 export const myFaresEnabled = devOrTest;
 export const saveProductsEnabled = true;
 export const exportEnabled = false;
-export const masterStopListEnabled = dev;
+export const masterStopListEnabled = false;
```
Comment (JwanKhalaf) [inline repos/fdbt-site/src/constants/featureFlag.ts:9]: Do we want to keep this as `false`?

File: repos/fdbt-site/src/data/auroradb.ts [hunk 2]
```diff
@@ -1202,6 +1202,34 @@ export const getGroupPassengerTypesFromGlobalSettings = async (nocCode: string):
     }
 };
 
+export const getPassengerTypeNameByIdAndNoc = async (id: number, nocCode: string): Promise<string> => {
```
Comment (jacsamell) [inline repos/fdbt-site/src/data/auroradb.ts:1205]: Does this not exist already? Thought J added something like this? Alternatively you could just use the getPassengerTypeById and just use the name?

File: repos/fdbt-site/tsconfig.json [hunk 3]
```diff
@@ -24,7 +24,8 @@
     "rootDir": "./",
     "jsx": "preserve",
     "sourceMap": true,
-    "baseUrl": "."
+    "baseUrl": ".",
+    "useUnknownInCatchVariables": false,
```
Comment (JwanKhalaf) [inline repos/fdbt-site/tsconfig.json:28]: What does this do?

File: repos/fdbt-site/tests/pages/products/productDetails.test.tsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        it('correctly returns the elements which should be displayed on the page for a school single ticket', async () => {
+            (getProductsMatchingJson as jest.Mock).mockResolvedValueOnce(expectedSingleTicket);
+            const ctx = getMockContext({ query: { productId: '1' } });
+            expect(await getServerSideProps(ctx)).toStrictEqual({
+                props: {
+                    productName: 'Test Passenger Type - Single (school)',
+                    startDate: '17/12/2020',
+                    endDate: '18/12/2020',
+                    productDetailsElements: [
+                        { name: 'Service', content: '215 - Worthing - Seaham - Crawley' },
```
Comment (JwanKhalaf) [inline repos/fdbt-site/tests/pages/products/productDetails.test.tsx:71]: Should `journey direction` be among these?

File: repos/fdbt-site/src/interfaces/typeGuards.ts [hunk 5]
```diff
@@ -176,3 +183,6 @@ export const isProductInfo = (
 export const isPeriodExpiry = (
     periodExpiryAttribute: PeriodExpiry | ErrorInfo[] | undefined,
 ): periodExpiryAttribute is PeriodExpiry => !!periodExpiryAttribute && 'productValidity' in periodExpiryAttribute;
+
+export const isBasePeriodTicket = (ticket: WithIds<Ticket>): ticket is WithIds<BasePeriodTicket> =>
```
Comment (jacsamell) [inline repos/fdbt-site/src/interfaces/typeGuards.ts:187]: Do we need this?

File: repos/fdbt-site/src/pages/products/productDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+    endDate,
+    productDetailsElements,
+}: ProductDetailsProps): ReactElement => (
+    <TwoThirdsLayout title={title} description={description} errors={[]}>
+        <h1 className="govuk-heading-l">{productName}</h1>
+        <div id="contact-hint" className="govuk-hint">
+            Product status: {getTag(startDate, endDate)}
+        </div>
+        {productDetailsElements.map((element) => {
+            const content = isArray(element.content) ? element.content : [element.content];
```
Comment (jacsamell) [inline repos/fdbt-site/src/pages/products/productDetails.tsx:37]: FYI there is a toArray method in utils

File: repos/fdbt-site/src/pages/products/productDetails.tsx [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+
+    if ('journeyDirection' in ticket && ticket.journeyDirection) {
+        productDetailsElements.push({ name: 'Journey direction', content: ticket.journeyDirection });
+    }
+
+    const passengerTypeName = await getPassengerTypeNameByIdAndNoc(ticket.passengerType.id, noc);
+    productDetailsElements.push({ name: 'Passenger type', content: passengerTypeName });
+
+    const isSchoolTicket = 'termTime' in ticket && ticket.termTime;
+    if (!isSchoolTicket) {
```
Comment (jacsamell) [inline repos/fdbt-site/src/pages/products/productDetails.tsx:82]: Do school tickets not have Time Restrictions?

File: repos/fdbt-site/src/data/auroradb.ts (hunk 2 above)
Comment (matt-twomey) [reply repos/fdbt-site/src/data/auroradb.ts:1205]: My bad, I think we hadn't pulled J's changes when we wrote this - will get rid!

File: repos/fdbt-site/src/interfaces/typeGuards.ts (hunk 5 above)
Comment (matt-twomey) [reply repos/fdbt-site/src/interfaces/typeGuards.ts:187]: We do not. Removed 👍

**Pull request merged**